src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -752,6 +752,35 @@ internal static bool TryParseAsConstantHashtable(string input, out Hashtable res
             return true;
         }
 
+        // Collect types that keep .resx file strings
+        // for errors thrown with "SaveError"
+        private static readonly Type[] s_resxTypes = new []
+        {
+            typeof(ParserStrings),
+            typeof(ExtendedTypeSystem),
+            typeof(DiscoveryExceptions)
+        };
+
+        /// <summary>
+        /// Given an errorId and an errorMsg, verify that they correspond to one another
+        /// in some .resx file
+        /// </summary>
+        /// <param name="errorId">the ID (or RESX key) of the error</param>
+        /// <param name="errorMsg">the message (or RESX value) of the error</param>
+        /// <returns>true if the ID and message are paired in some .resx file, false otherwise</returns>
+        private static bool ErrorIdCorrespondsToMsg(string errorId, string errorMsg)
+        {
+            foreach (Type resxType in s_resxTypes)
+            {
+                if (errorMsg.Equals(resxType.GetProperty(errorId)?.GetValue(null)))
```

Comment:
> resxType.GetProperty(errorId)

The properties in `resxType` are all internal, so this won't get back anything.

Reply:
> I'd consider moving the resources instead of writing a helper function.

I tend to agree with Jason's comment. As for the two "ParserError" instances, I don't mind have the resource string ("ParserError" => "{0}") in `ParserStrings`.

Reply:
Ok cool. I'll do that